controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller_test.go

```diff
@@ -67,7 +76,33 @@ func TestSpaceProvisionerConfigValidation(t *testing.T) {
 		assert.NoError(t, reconcileErr)
 		AssertThat(t, spc, Is(NotReadyWithReason(toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason)))
 	})
-	t.Run("becomes valid when the referenced ToolchainCluster appears", func(t *testing.T) {
+	t.Run("is not ready when existing not-ready ToolchainCluster is referenced", func(t *testing.T) {
+		// given
+		spc := NewSpaceProvisionerConfig("spc", test.HostOperatorNs, ReferencingToolchainCluster("cluster1"))
+		r, req, cl := prepareReconcile(t, spc, &toolchainv1alpha1.ToolchainCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cluster1",
+				Namespace: test.HostOperatorNs,
+			},
+			Status: toolchainv1alpha1.ToolchainClusterStatus{
+				Conditions: []toolchainv1alpha1.ToolchainClusterCondition{
+					{
+						Type:   toolchainv1alpha1.ToolchainClusterReady,
+						Status: v1.ConditionFalse,
+					},
+				},
+			},
+		})
+
+		// when
+		_, reconcileErr := r.Reconcile(context.TODO(), req)
+		require.NoError(t, cl.Get(context.TODO(), runtimeclient.ObjectKeyFromObject(spc), spc))
+
+		// then
+		assert.NoError(t, reconcileErr)
+		AssertThat(t, spc, Is(NotReadyWithReason(toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotReadyReason)))
+	})
+	t.Run("becomes ready when the referenced ToolchainCluster appears and is ready", func(t *testing.T) {
```

Comment:
should this be a subtest of the previous one ( "is not ready when existing not-ready ToolchainCluster is referenced " ) and use the same `spc` variable so that we check that a second reconcile would put that back to ready when toolchaincluster gets ready ?